pulp_rpm/tests/performance/test_sync.py

```diff
@@ -152,3 +166,86 @@ def test_centos8_kickstart_baseos_on_demand(self):
     def test_centos8_kickstart_appstream_on_demand(self):
         """Kickstart Sync CentOS 8 AppStream."""
         self.rpm_sync(url=CENTOS8_KICKSTART_APP_URL)
+
+
+class CDNTestCase(unittest.TestCase):
+    """Sync a repository from CDN."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = gen_rpm_client()
+        cls.repo_api = RepositoriesRpmApi(cls.client)
+        cls.remote_api = RemotesRpmApi(cls.client)
+        cls.repometadatafiles = ContentRepoMetadataFilesApi(cls.client)
+        delete_orphans(cls.cfg)
+        # Certificates processing
+        cls.cdn_client_cert = False
+        if os.environ['CDN_CLIENT_CERT'] \
+                and os.environ['CDN_CLIENT_KEY'] \
+                and os.environ['CDN_CA_CERT']:
+            # strings have escaped newlines from environmental variable
+            cls.cdn_client_cert = os.environ['CDN_CLIENT_CERT'].replace('\\n', '\n')
+            cls.cdn_client_key = os.environ['CDN_CLIENT_KEY'].replace('\\n', '\n')
+            cls.cdn_ca_cert = os.environ['CDN_CA_CERT'].replace('\\n', '\n')
+
+    @skip_if(bool, "cdn_client_cert", False)
+    def test_sync_with_certificate(self):
+        """Test sync against CDN.
+
+        1. create repository, appstream remote and sync
+            - remote using certificates and tls validation
+        2. create baseos remote and sync to same repository
```

Comment:
Could we sync into 2 different repos, as a more common use case?
Even though I agree that the current issue will be tested in both scenarios, syncing into one repo has different behaviour, and checks for duplicates are in place.

If we cover a more common use case, we have higher chances to catch some other issues during the sync.

Reply:
Now using own repo for each remote. And added check if each repo has its own 'productid'.